greedy_max/__init__.py

```diff
@@ -532,39 +675,77 @@ def schedule(self):
-                                                                          obs_order[i], i_start[i], i_end[i],
-                                                                          np.max(abs(
-                                                                              self.time_slots.weights[site][
-                                                                                  obs_order[i]][
-                                                                                  i_start[i]:i_end[i] + 1]))))
-                        sum_score += np.sum(abs(self.time_slots.weights[site][obs_order[i]][i_start[i]:i_end[i] + 1]))
-                        time_used += (i_end[i] - i_start[i] + 1)
-            # input()
-
-        print('Sum score = {:7.2f}'.format(sum_score))
-        print('Sum score/time step = {:7.2f}'.format(sum_score / (2 * self.time_slots.total)))
-        print('Time scheduled = {:5.2f}'.format(time_used * self.time_slots.slot_length.to(u.hr)))
+                obs_in_plan = self.plan.get_observation_orders(site)
+
+                output_table = []
+
+                for observation in obs_in_plan:
+                    obs_idx = observation[0]
+                    start = observation[1]
+                    end = observation[2]   
+                    if obs_idx != EMPTY and obs_idx != UNSCHEDULABLE:
+                        unit = self.plan.get_unit_by_observation(site, obs_idx) # TODO: handle None case
+                        name = all_obs[obs_idx].name
+                        weights = np.max(abs(self.time_slots.weights[site][unit.idx][start:end+1]))
+                        category = all_obs[obs_idx].category.name
+                        output_table.append([name,obs_idx,category,start, end, weights])
+                        sum_score += np.sum(abs(self.time_slots.weights[site][unit.idx][start:end+1]))
+                        time_used += (end - start + 1)    
+                print(tabulate(output_table, headers=['Obs', 'obs_order', 'category','start', 'end', 'Max W']))
+            input()
 
+        print(f'Sum score = {sum_score:7.2f}')
+        print(f'Sum score/time step = {(sum_score / (len(self.sites) * self.time_slots.total)):7.2f}') 
+        print(f'Time scheduled = {(time_used * self.time_slots.slot_length.to(u.hr)):5.2f}')
+       
         return self
+
+    def print_plan(self):
+        
+        for site in self.sites:
+            # if s == 'gs':
+            #     night_length = gs_night_length[i_day]
+            # else:
+            #     night_length = gn_night_length[i_day]
+            print(Site(site).name.upper())
+            obs_order, i_start, i_end = get_order(plan=self.plan[site])
+            # print(obs_order, i_start, i_end)
+            sum_score = 0.0
+            sum_metric = 0.0
+            time_used = 0.0 * u.hr
+            for i in range(len(obs_order)):
+                if obs_order[i] >= 0:
```

Comment:
That is the old version of the output of GreedyMax, which is basically the same as the current one. Most of that code won't work so it's better to delete it but I think it might be necessary to define the structure of the output we want from GreedyMax.